File: frontend/src/api/jobs.ts
import { request } from './client'

export interface JobSearchFilters {
  keywords: string
  location?: string
  workplace?: string
  experience?: string
  jobType?: string
  postedWithin?: string
}

export interface JobSearchURL {
  provider: string
  url: string
}

export function getSearchURL(filters: JobSearchFilters) {
  const params = new URLSearchParams()
  for (const [key, value] of Object.entries(filters)) {
    if (value) params.set(key, value)
  }
  return request<JobSearchURL>(`/api/jobs/search-url?${params.toString()}`)
}

File: backend/internal/handlers/jobs.go
package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"istream/backend/internal/jobsearch"
)

// handleJobSearchURL builds a pre-filtered LinkedIn jobs-search deep link
// from the user's filters.
func (s *Server) handleJobSearchURL(c *gin.Context) {
	params := jobsearch.SearchParams{
		Keywords:     c.Query("keywords"),
		Location:     c.Query("location"),
		Workplace:    c.Query("workplace"),
		Experience:   c.Query("experience"),
		JobType:      c.Query("jobType"),
		PostedWithin: c.Query("postedWithin"),
	}
	url, err := jobsearch.BuildLinkedInURL(params)
	if err != nil {
		errorJSON(c, http.StatusBadRequest, err.Error())
		return
	}
	c.JSON(http.StatusOK, gin.H{"provider": "linkedin", "url": url})
}
